File: src/pi.kt
fun calculatePi(precision: Int): Double {
  var (s: Double, k: Double) = Pair(0.0, 1.0)
  var max: Int = Math.pow(10.0, precision.toDouble()).toInt()
  for (i in 0 until max) {
    if (i % 2 == 0) 
      s += 4 / k
    else 
      s -= 4 / k
    k += 2
  }
  return s
}

fun main() {
  var pi: Double = calculatePi(9)
  println("Pi = %.8f".format(pi))
}

File: src/pi.swift
import Foundation

func calculatePi(precision: Int) -> Double {
  var s = 0.0, k = 1.0
  let max = Int(pow(10.0, Double(precision)))
  for i in 0..<max {
    if i % 2 == 0 {
      s += 4 / k
    } else {
      s -= 4 / k
    }
    k += 2
  }
  return s
}

var pi = calculatePi(precision: 9)
print(String(format: "Pi = %.8f", pi))
